sklearn/impute/_base.py

```diff
@@ -717,6 +740,25 @@ def _fit(self, X, y=None):
 
         return missing_features_info[0]
 
+    def _fit_precomputed(self, X):
```

Comment:
I think I am still not fully convinced of this. I do see the comment here: https://github.com/scikit-learn/scikit-learn/pull/15089#discussion_r334258979

But since it is private, I think it would not have too much impact.